executions/graphql-kotlin-transaction-batcher/src/main/kotlin/com/expediagroup/graphql/transactionbatcher/publisher/TriggeredPublisher.kt

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2022 Expedia, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.expediagroup.graphql.transactionbatcher.publisher
+
+import com.expediagroup.graphql.transactionbatcher.transaction.BatcheableTransaction
+import com.expediagroup.graphql.transactionbatcher.transaction.TransactionBatcherCache
+import org.reactivestreams.Publisher
+import org.reactivestreams.Subscriber
+import org.reactivestreams.Subscription
+import org.slf4j.LoggerFactory
+
+/**
+ * Interface representing a publisher with input [TInput] type and output [TOutput] type
+ */
+@Suppress(
+    "ReactiveStreamsSubscriberImplementation",
```

Comment:
why are we suppressing this warning? is this valid?

Reply:
it is valid, given that `Subscriber` is an interface, so it's valid to implement it, However a warning appears in the IDE when trying to do it.

i can see a lot of `Subscriber` implementations that need to add that suppress as well
https://github.com/search?p=2&q=%22ReactiveStreamsSubscriberImplementation%22&type=Code

Reply:
unsure, I am not getting any warning in my local

Reply:
<img width="583" alt="image" src="https://user-images.githubusercontent.com/6611331/155080511-6bb16adf-02fe-498f-8bbe-407bfff65f0d.png">